src/main/java/org/folio/dcb/service/impl/DcbHubLocationServiceImpl.java

```diff
@@ -43,43 +53,68 @@ public class DcbHubLocationServiceImpl implements DcbHubLocationService {
   private final DcbHubLocationClient dcbHubLocationClient;
   private final LocationsClient locationsClient;
   private final LocationUnitClient locationUnitClient;
+  private final InventoryServicePointClient servicePointClient;
+  private final DcbHubProperties dcbHubProperties;
 
   @Value("${application.dcb-hub.batch-size}")
   private Integer batchSize;
 
   @Override
-  public void createShadowLocations(ServicePointRequest servicePointRequest) {
+  public RefreshShadowLocationResponse createShadowLocations() {
     try {
+      log.debug("createShadowLocations:: creating shadow locations");
+
+      if (!Boolean.TRUE.equals(dcbHubProperties.getFetchDcbLocationsEnabled())) {
+        log.info("createShadowLocations:: DCB Hub locations fetching is disabled, skipping shadow location creation");
+        throw new DcbHubLocationException("DCB Hub locations fetching is disabled, skipping shadow location creation", HttpStatus.BAD_REQUEST);
+      }
+
+      List<ServicePointRequest> servicePointList = servicePointClient.getServicePointByName(NAME).getResult();
+      if (servicePointList.isEmpty()) {
+        log.info(
+          "createShadowLocations:: No service points found with name {}, So cannot create shadow locations without {} service point",
+          NAME, NAME);
+        throw new DcbHubLocationException(String.format("No service points found with name %s, So cannot create shadow locations without %s service point", NAME, NAME), HttpStatus.NOT_FOUND);
+      }
+
+      ServicePointRequest servicePointRequest = servicePointList.getFirst();
       log.info("createShadowLocations:: fetching all locations from DCB Hub");
+
       List<DcbHubLocationResponse.Location> locationList = fetchDcbHubAllLocations();
       if (CollectionUtils.isEmpty(locationList)) {
         log.info("createShadowLocations:: No locations found in DCB Hub, skipping shadow location creation");
-        return;
+        throw new DcbHubLocationException("No locations found in DCB Hub, skipping shadow location creation", HttpStatus.NOT_FOUND);
       }
-      createShadowLocations(locationsClient, locationUnitClient, servicePointRequest, locationList);
+
+      RefreshShadowLocationResponse refreshShadowLocationResponse = new RefreshShadowLocationResponse()
+        .locationUnits(new RefreshShadowLocationResponseLocationUnits());
+      createShadowLocations(servicePointRequest, locationList, refreshShadowLocationResponse);
```

Comment:
Use a functional approach here, and instead of using the void method -> return a list of created shadow locations and set them into the response